proxy/json.go

```diff
@@ -27,12 +28,19 @@ func lookupObject(m map[string]interface{}, key string) (map[string]interface{},
 		return nil, &UnmarshalWrongTypeError{key, "object", iface}
 	}
 
-	return result, nil
+	return jsonObject(result), nil
 }
 
-// For parsing json objects. Look up (or create) a string in the given map.
-func lookupString(m map[string]interface{}, key string) (string, error) {
-	iface, ok := m[key]
+func (j jsonObject) String(key string, ks ...string) (string, error) {
+	if len(ks) > 0 {
+		j, err := j.Object(key)
+		if err != nil {
+			return "", err
+		}
+		return j.String(ks[0], ks[1:]...)
+	}
+
+	iface, ok := j[key]
```

Comment:
As noted before, it is odd that this method performs a traversal, whereas `Object` and `StringArray` don't.  Seems to me that it would be trivial to traverse on all of them - the traversal code could live in `Object` alone, or some helper fun.